src/structures/mod.rs

```diff
@@ -25,37 +25,37 @@ pub use names::name::Name;
 mod buffers;
 pub use self::auth_buffer::Auth;
 pub mod auth_buffer {
-    pub use super::buffers::auth::*;
+    pub use super::buffers::Auth;
```

Comment:
These make no sense now. They should be removed.

Reply:
Should it just be `pub mod buffers` then?

Reply:
Yes. Because previously the xxx_buffer mod was used in order to access extra things that had not been declared pub use.

The macro could add each buffer type in its own pub mod. So that it would be possible to add stuff to the macro without having to explicitly declare it with pub use in order to make it available.

Reply:
I've simply made them public from the `structures` module - reckon the modules themselves are useful?

Reply:
The only reason I can see for keeping the modules is the following is if you wish to add like free functions or something similar where the name does not change depending buffer type. Or if you add free functions with the same name but different return types. That won't work unless they are 1. Generic or 2. in different modules.

Reply:
I am actually starting to lean against that it would be a good idea to keep the different buffer types in different modules. That way you can leave the src/structures/buffers/mod.rs unchanged and just add

```rust
macro_rules! buffer_type {
    ($module_name:idnet,$native_type:ident,$MAX:expr,$tss_type:ident) => {
        pub mod $module_name {
            use super::*;
            use crate::tss2_esys::$tss_type;
```

Reply:
I don't think it would work that way since it would be impossible to add things in the module that are not in the macro. I'll just wrap each macro call in a module.

Reply:
Will not recreate the previous export hierarchy, though - will only export the types from `structures/mod.rs`, not the modules as well, since they only contain the type.

Reply:
Yeah, I think that is fine. I do not think we have a reason for having the previous export structure so removing it is fine and each type is confined in its own mod with its own imports which is nice to.